Tests/AzurePSDrive.Tests.ps1

```diff
@@ -7,10 +7,12 @@ param (
     [string]$subscriptionName = 'AutomationTeam'
 )
 
+function New-PartialGuidForName { ([guid]::NewGuid().ToString() -replace '-','')[0..9] -join '' }
+
 #region Script variables
-$resourceGroupName = 'AzurePSDrive.Test'
+$resourceGroupName = "rgapsdt$(New-PartialGuidForName)"
 $location = 'WestUS'
-$storageAccountName = 'azurepsdriveteststorage'
+$storageAccountName = "saapsdt$(New-PartialGuidForName)"
```

Comment:
Are the ResourceGroup and StorageAccount autogenerated every single time? 

Since these are static resources, we want to create them once and reuse for every test run. Ideally Initialize-TestEnv must happen only once for a given Account..

Reply:
@raghushantha I set up the tests so that you could run them on any account, as long as you provide a service principal. This is very useful so that no pre-configured environment is necessary, the tests just work. Additionally, I made it so that when the tests complete, they clean up the resources they created, so they don't persist on the account. 

Having a persisted environment introduces 'assumptions' about the environment, and can cause problems. So, for example, in your very last test, you had code checking to see that a blob named 'foo' existed, but none of the tests created 'foo'. So, that is an assumption about the environment that isn't apparent to someone running the tests. Instead, I made the checks validate that nothing is in the environment, since that's the state it would be in if it is created during every test run. Similarly, you could have 'foo' created as part of the setup, and then check that assumption.

I'm do this similar process for testing our cmdlets in our CI environment. Creation -> Tests -> Removal. It makes debugging easier since you can assume the environment is correct, unless the environment initialization fails for some reason. Then, you'd have that information as part of the test process.

Additionally, the reason I changed only the resource group and storage account names is because they are global names. Meaning, they have to be unique among **all** subscriptions. Everything else can have static names and they'll work fine, as they only need to be unique to that resource group.